pkg/apis/aws/validation/shoot.go

```diff
@@ -91,6 +91,27 @@ func ValidateWorkersUpdate(oldWorkers, newWorkers []core.Worker, fldPath *field.
 	return allErrs
 }
 
+func ValidateNetworkingUpdate(oldNetworkConfig, networkConfig *core.Networking, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	allErrs = append(allErrs, ValidateIPFamiliesUpdate(oldNetworkConfig.IPFamilies, networkConfig.IPFamilies, fldPath.Child("ipFamilies"))...)
+	return allErrs
+}
+
+func ValidateIPFamiliesUpdate(oldIPFamilies, newIPFamilies []core.IPFamily, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if (len(oldIPFamilies) == 0 && len(newIPFamilies) == 0) ||
+		(len(oldIPFamilies) == 0 && len(newIPFamilies) == 1 && newIPFamilies[0] == core.IPFamilyIPv4) ||
+		(len(oldIPFamilies) == 1 && len(newIPFamilies) == 1 && oldIPFamilies[0] == newIPFamilies[0]) ||
+		(len(oldIPFamilies) == 2 && len(newIPFamilies) == 2) ||
+		(len(oldIPFamilies) == 1 && oldIPFamilies[0] == core.IPFamilyIPv4 && len(newIPFamilies) == 2 && newIPFamilies[0] == core.IPFamilyIPv4 && newIPFamilies[1] == core.IPFamilyIPv6) {
+		return allErrs
+	}
```

Comment:
Please prepend with a `slices.Equals(...)` check to remove the simple equality cases and have only real changes in the condition.